static/app/views/projectEventRedirect.tsx

```diff
@@ -50,10 +50,17 @@ function ProjectEventRedirect({router}: Props) {
       // responseURL is the URL of the document the browser ultimately loaded,
       // after following any redirects. It _should_ be the page we're trying
       // to reach; use the router to go there.
-
+      //
       // Use `replace` so that hitting the browser back button will skip all
       // this redirect business.
-      router.replace(xhr.responseURL);
+      const url = new URL(xhr.responseURL);
+      if (url.origin === window.location.origin) {
+        router.replace(url.pathname);
+      } else {
+        // If the origin has changed, we cannot do a simple replace with the router.
+        // Instead, we opt to do a full redirect.
+        window.location.replace(xhr.responseURL);
```

Comment:
This doesn't work yet until the right CORs headers are set

Reply:
What isn't correct with cors headers currently?

Reply:
After debugging this more closely, this isn't where the CORs error is happening. The only way I was able to reproduce this CORs error is by using webpack dev server to reroute the 302 to go to a different subdomain.

e.g.
1. start on `org1.sentry.io/org1/proj1/events/abcdef`
2. force the response to the HEAD request to redirect to `org2.sentry.io/issues/123/`
3. the HEAD request will follow the redirect and make a preflight request where the OPTIONS request is missing all the CORS headers which i believe is due to [this](https://github.com/getsentry/sentry/blob/f24f9d53e0bd1cd94a8850c9086a848fa618bef8/src/sentry/api/base.py#L116-L118) which skips all the CORS headers when it's an OPTIONS request

I'm not sure this is an actual case in practice.